ci/build.yml

```
@@ -413,6 +413,15 @@ jobs:
     - template: tell-slack-failed.yml
     - template: report-end.yml
 
+- job: compatibility_stable_protobuf
+  pool:
+    name: ubuntu_20_04
+    demands: assignment -equals default
+  steps:
+    - checkout: self
```

```suggestion
    - checkout: self
    - bash: ci/dev-env-install.sh
      displayName: 'Build/Install the Developer Environment'
```
I think this might be what results in the nix file lock

---

We have lots of jobs that don't do that and work fine. 🤔

---

Do those jobs rely on nix artifacts like this one does? Maybe it’s just very flaky but I reset one node and then it ran into the same issue on another node.

---

They do. What seems to be issue here is that we `git checkout` to another version of nixpkgs (Linux CI nodes run the `dev-env-install` on startup, so they should have all packages pre-fetched).

I'm not entirely sure how that breaks, but, do we need to re-run the `eval "$(dev-env/bin/dade-assist)"` after the `git checkout`? When I run `dade-assist` I don't see anything that looks like it would be version-specific, but when we get a new nixpkgs sometimes it helps if I `cd` out and back in.

---

Ah I think you’re right, we might need to rerun it.

---

Another option would be to revert to the `git archive` solution that doesn't touch the main work tree. Which one do you think is best?

---

I can reproduce locally and running `eval` again does not help.

What happens is that we have dev-env activated, and:

- We run `buf`, which activates the dev-env script.
- The dev-env script sees that `buf` is not the right version, so it triggers a nixpkgs fetch.
- Nix then needs to un-tar its nixpkgs definition, and because dev-env is activated, tries to go through dev-env `tar` for that.
- `tar` is a dev-env script, so it tries to build itself from Nix, which is already waiting with a lock in hand.

I'm not quite sure how do work around that. Without direnv I'm not quite sure how we properly _de_activate dev-env within a Bash script. 🤔 

@fabiotudone-da I'd like a solution to the above issue in principle, but yes, for this PR it may be faster to just go back to the git-archive approach. What I don't like about that approach is that we expect the HEAD script to know about the folder structure of "random" old versions of the repo, and that may lead to subtle breakages down the road.

---

> What I don't like about that approach is that we expect the HEAD script to know about the folder structure of "random" old versions of the repo, and that may lead to subtle breakages down the road.

I'll try to extract everything but not to move in there (or at least to avoid extracting the dev-env portions).

---

Just noticed your suggestion below, I'll try that first.

---

Yep (fba0582), this time [it only took half-a-minute](https://dev.azure.com/digitalasset/daml/_build/results?buildId=89412&view=logs&jobId=00351e57-e4ff-5cd1-1d0e-7baed9c3f219&j=00351e57-e4ff-5cd1-1d0e-7baed9c3f219&t=6bfdd20f-1850-50bd-a925-9faae3198299). Thanks!